packages/turf-point-grid/index.js

```diff
@@ -1,73 +1,72 @@
- * @param {number} cellSide the distance between points
+ * @param {Array<number>} bbox extent in [minX, minY, maxX, maxY] order
+ * @param {number} cellSide the distance between points, in units
  * @param {Object} [options={}] Optional parameters
- * @param {string} [options.units="kilometers"] used in calculating cellSide, can be degrees, radians, miles, or kilometers
- * @param {number} [options.bboxIsMask=false] if true, and bbox is a Polygon or MultiPolygon, the grid Point will be created
+ * @param {string} [options.units='kilometers'] used in calculating cellSide, can be degrees, radians, miles, or kilometers
+ * @param {Feature<Polygon|MultiPolygon>} [options.mask] if passed a Polygon or MultiPolygon, the grid Points will be created only inside it
  * @param {Object} [options.properties={}] passed to each point of the grid
  * @returns {FeatureCollection<Point>} grid of points
  * @example
  * var extent = [-70.823364, -33.553984, -70.473175, -33.302986];
  * var cellSide = 3;
+ * var options = {units: 'miles'};
  *
- * var grid = turf.pointGrid(extent, cellSide, {units: 'miles'});
+ * var grid = turf.pointGrid(extent, cellSide, options);
  *
  * //addToMap
  * var addToMap = [grid];
  */
 function pointGrid(bbox, cellSide, options) {
     // Optional parameters
     options = options || {};
-    if (typeof options !== 'object') throw new Error('options is invalid');
-    var bboxIsMask = options.bboxIsMask || false;
-    var properties = options.properties || {};
+    if (!isObject(options)) throw new Error('options is invalid');
+    var units = options.units;
+    var maskIsPoly = options.mask && (getType(options.mask) === 'Polygon' || getType(options.mask) === 'MultiPolygon');
+    var properties = (options.properties || {});
 
     // Containers
     var results = [];
-    var bboxMask = bbox;
 
     // validation
+    if (cellSide === null || cellSide === undefined) throw new Error('cellSide is required');
+    if (!isNumber(cellSide)) throw new Error('cellSide is invalid');
     if (!bbox) throw new Error('bbox is required');
-    if (!Array.isArray(bbox)) bbox = turfBBox(bbox); // Convert GeoJSON to bbox
+    if (!Array.isArray(bbox)) throw new Error('bbox must be array');
     if (bbox.length !== 4) throw new Error('bbox must contain 4 numbers');
-    if (!isObject(options)) throw new Error('options is invalid');
 
-    var  west = bbox[0];
-    var  south = bbox[1];
-    var  east = bbox[2];
-    var  north = bbox[3];
+    var west = bbox[0];
+    var south = bbox[1];
+    var east = bbox[2];
+    var north = bbox[3];
 
-    var  xFraction = cellSide / (distance(point([west, south]), point([east, south]), options));
-    var  cellWidth = xFraction * (east - west);
-    var  yFraction = cellSide / (distance(point([west, south]), point([west, north]), options));
-    var  cellHeight = yFraction * (north - south);
+    var xFraction = cellSide / (distance(point([west, south]), point([east, south]), units));
```

Comment:
Distance no longer requires the input data to be Point, it can be an Array of numbers.

```js
distance([west, south], [east, south], options)
```

Also, all modules have changed to the `options` param syntax, even the ones with 1 param.